### DISCUSSION TITLE
Expansion of the PWM channels from 5 to 7 in Tasmota

### DISCUSSION THREAD

**Discussion by Basti149** [arendst/Tasmota#10626] [Ideas]
Schön guten Abend. Tasmota ist einfach Klasse und einfach zu bedienen.
Ich bräuchte für mein neues Projekt mehr als 5 PWM Kanäle. Könnte man ca. 2 weitere Kanäle (weiß Licht mit einfachen Regler würde zureichen) in die nächste Tasmota Version integrieren. Und zwar in die EP32.bin da dieser bekanntlich mehr als 5 PWM Kanäle zulassen würde.

Mit freundlichen Grüßen Basti

Good evening. Tasmota is just great and easy to use.
I would need more than 5 PWM channels for my new project. If you could integrate about 2 more channels (white light with a simple controller would be sufficient) in the next Tasmota version. Namely in the EP32.bin as it is known that this would allow more than 5 PWM channels.

Sincerely, Basti

**Comment by vic42** [comment]
PWM is done in software so the number of PWM channels is not a hardware limitation per se. Adding more channels may severely complicate the lights module so I don't think it is going to happen anytime soon. But who knows.
A solution could be adding a second ESP or an i2c PWM chip like the PCA9685 (https://tasmota.github.io/docs/PCA9685/), depending on the application.

**Comment by thermseekr** [comment]
Hello, has anything evolved in this subject? I too would like to have more than 5 PWMs on ESP32, for driving simple white led stripes. The ideal count for me would be 8...

**Reply by barbudor** [reply]
There's a great file called [CHANGELOG.md](https://github.com/arendst/Tasmota/blob/development/CHANGELOG.md)
where you could find that ESP32 now supports 16 PWM since Feb 4th, v10.1.06

**Reply by sfromis** [reply]
You can have a regular Tasmota light (rgbcw) on PWM1-5 and still control PWM6-16 yourself, meaning not as a light.

**Reply by thermseekr** [reply]
I had v10.0 in the device I was using to test, so I was only seeing 5 PWMs. Problem solved, thank you very much for replying.
